apps/hash-api/src/graphql/resolvers/knowledge/file/request-file-upload.ts

```diff
@@ -7,7 +8,11 @@ import {
   ResolverFn,
 } from "../../../api-types.gen";
 import { LoggedInGraphQLContext } from "../../../context";
-import { dataSourcesToImpureGraphContext } from "../../util";
+import { graphQLContextToImpureGraphContext } from "../../util";
+
+const maximumFileSizeInMegaBytes = 100;
+
+const maximumFileSizeInBytes = maximumFileSizeInMegaBytes * 1024 * 1024;
```

Comment:
Let's comment inline why we're doing this, so that we recall the reasoning (and don't just remove the cap when we need to support large videos, etc in the future -- without creating a different process for handling those larger files).

Reply:
I've added a comment in https://github.com/hashintel/hash/pull/3929/commits/8c4f10b91d1350a8a398df0e5f94c4361d3c094e, I'm realising the limitation I recall technically only applies to the files which are parsed in temporal. If we feel strongly about not having a limit, then we could also just trigger the temporal workflow for files larger than X, or just enforce the upload size for the file type's which would trigger a temporal workflow.

Or was there another reason we wanted to add a maximum file size I'm forgetting about?

Reply:
Not that I'm aware of. @CiaranMn?

Reply:
Given that the entire text content is saved as a property on the entity, it is also a memory issue for the database and anywhere we pass the entity around, depending on how big the strings are derived from these files. Really we don't want entities with 10MB properties in them, let alone 100MB or anything greater. This won't scale (imagine querying and viewing a list of documents that all have huge strings in them) and we need to store the text content some other way before any real usage.